datafusion/optimizer/src/decorrelate_predicate_subquery.rs

```diff
@@ -1407,13 +1361,13 @@ mod tests {
             .build()?;
 
         let expected = "Projection: customer.c_custkey [c_custkey:Int64]\
-                        \n  LeftSemi Join:  Filter: __correlated_sq_1.o_custkey = customer.c_custkey [c_custkey:Int64, c_name:Utf8]\
```

Comment:
This change is because I fixed how the rule first rewrites nested subqueries first. The old logic did not traverse the nested subquery but instead just tried to rewrite the first node, which did not really work since it would be a Subquery node not a filter.

Reply:
The changes are that the auto generated subquery alias (e.g. `__correlated_sq_1`) is a different order, right?  Or are you referring to something more substantial?

Reply:
No, only the order the alias is assigned.